ray-curator/ray_curator/stages/video/clipping/clip_extraction_stages.py

```diff
@@ -0,0 +1,386 @@
+from ray_curator.tasks import Clip, Video, VideoTask
+from ray_curator.utils import grouping
+from ray_curator.utils.operation_utils import make_pipeline_temporary_dir
+
+
+@dataclass
+class ClipTranscodingStage(ProcessingStage[VideoTask, VideoTask]):
+    """Stage that transcodes video clips into a standardized format.
+
+    This stage handles the conversion of video clips using FFmpeg, supporting both
+    software (libx264, libopenh264) and hardware (NVENC) encoding with configurable parameters.
+
+    Args:
+        num_cpus_per_worker: Number of CPUs per worker.
+        encoder: Video encoder to use.
+        encoder_threads: Number of threads per encoder.
+        encode_batch_size: Number of clips to encode in parallel.
+        nb_streams_per_gpu: Number of streams per GPU.
+        use_hwaccel: Whether to use hardware acceleration.
+        use_input_bit_rate: Whether to use input video bit rate.
+        num_clips_per_chunk: Number of clips per chunk. If the number of clips is larger than this, the clips will be split into chunks, and created VideoTasks for each chunk.
+        verbose: Whether to print verbose logs.
+        ffmpeg_verbose: Whether to print FFmpeg verbose logs.
+    """
+    num_cpus_per_worker: float = 6.0
+    encoder: str = "libx264"
+    encoder_threads: int = 1
+    encode_batch_size: int = 16
+    nb_streams_per_gpu: int = 3
+    use_hwaccel: bool = False
+    use_input_bit_rate: bool = False
+    num_clips_per_chunk: int = 32
+    ffmpeg_verbose: bool = False
+    verbose: bool = False
+
+    @property
+    def name(self) -> str:
+        return "clip_transcoding"
+
+    def setup(self, worker_metadata: WorkerMetadata | None = None) -> None:  # noqa: ARG002
+        """Setup method called once before processing begins.
+        Override this method to perform any initialization that should
+        happen once per worker.
+        Args:
+            worker_metadata (WorkerMetadata, optional): Information about the worker (provided by some backends)
+        """
+        if self.encoder not in {"libopenh264", "libx264", "h264_nvenc"}:
+            error_msg = f"Expected encoder of `libopenh264`, `libx264`, or `h264_nvenc`. Got {self.encoder}"
+            raise ValueError(error_msg)
+
+    @property
+    def resources(self) -> Resources:
+        """Resource requirements for this stage."""
+        if self.encoder == "h264_nvenc" or self.use_hwaccel:
+            if self.nb_streams_per_gpu > 0:
+                # Assume that we have same type of GPUs
+                gpu_info = _get_local_gpu_info()[0]
+                nvencs = _make_gpu_resources_from_gpu_name(gpu_info.name).num_nvencs
+                gpu_memory_gb = _get_gpu_memory_gb()
+                return Resources(nvencs=nvencs // self.nb_streams_per_gpu, gpu_memory_gb=gpu_memory_gb // self.nb_streams_per_gpu)
+            else:
+                return Resources(gpus=1)
+
+        return Resources(cpus=self.num_cpus_per_worker)
```

Comment:
Let's not override properties but in __post_init__ override the `self._resources`. If you override properties we'll end up with a weird case if someone does `ClipTranscodingStage.with(resources=???)`